junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionUtils.java

```diff
@@ -36,6 +36,14 @@ static void fail(String message) {
 		fail(() -> message);
 	}
 
+	static void fail(String message, Throwable cause) {
```

Comment:
This is quite a new use of `fail` to me, so I think the method would be better named `failWithCause` or something along those lines to help prevent confusion.